src/wmtk/io/HDF5Reader.hpp

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include "MeshReader.hpp"
+
+#include <wmtk/Primitive.hpp>
+
+#include <filesystem>
+#include <vector>
+
+namespace wmtk {
+
+class Mesh;
+
+class HDF5Reader : public MeshReader
```

Comment:
We need documentation for all of these classes but that can be done in a separate PR.